src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java

```diff
@@ -43,7 +44,12 @@ public VendorResponseVO findByTitle(String title) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public VendorResponseVO create(VendorRequestVO request) {
-        return vendorMapper.toVO(vendorRepository.save(vendorMapper.toEntity(request)));
+        Optional<Vendor> byEmail = vendorRepository.findByEmail(request.getEmail());
+        if (byEmail.isPresent()) {
+            throw new EntityAlreadyExistsException("Email already exist");
```

Comment:
Let's move this logic to VendorValidator (new class responsible for entity validation)

Reply:
moved to separate class

Reply:
throwing exception should be in validator as wee

Reply:
moved to validator